extensions/2.0/Vendor/EXT_feature_metadata/schema/featureClass.property.schema.json

```diff
@@ -0,0 +1,98 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema",
+  "title": "Property class",
+  "type": "object",
+  "description": "A property class.",
+  "allOf": [ { "$ref": "glTFProperty.schema.json" } ],
+  "properties": {
+      "name": {
+          "type": "string",
+          "minLength" : 1,
+          "description": "The name of the property, e.g. for display purposes. The string is UTF-8 encoded."
+      },
+      "semantic": {
+          "type": "string",
+          "minLength": 1,
+          "description": "Identifies a property with a well-known meaning. The string is UTF-8 encoded. Semantics defined by this extension include `NAME` and `ID`"
+      },
+      "description": {
+          "type": "string",
+          "minLength" : 1,
+          "description": "The description of the property. The string is UTF-8 encoded."
+      },
+      "elementType": {
+          "type": "string",
+          "enum": [
+              "BOOLEAN",
+              "INT8",
+              "UINT8",
+              "INT16",
+              "UINT16",
+              "INT32",
+              "UINT32",
+              "INT64",
+              "UINT64",
+              "FLOAT16",
+              "FLOAT32",
+              "FLOAT64",
+              "BLOB",
+              "STRING",
+              "ARRAY",
+              "VARIABLE_SIZE_ARRAY"
+          ],
+          "description": "The type of each element of the property array. An `elementType` is either a basic type used for `componentType`, an `ARRAY`, or a `VARIABLE_SIZE_ARRAY`. If `ARRAY` or `VARIABLE_SIZE_ARRAY` is used, then `componentType` must also be specified. `ARRAY` is a fixed-length array, and requires a `componentsPerElement` property. `VARIABLE_SIZE_ARRAY`, `BLOB` and `STRING` are variable-length unless the feature table has `elementByteLength` defined, in which case values must be exactly this length. When `featureTable.elementByteLength` is not specified, `VARIABLE_SIZE_ARRAY`, `STRING`, and `BLOB` also require a corresponding `offsetBufferView`. This is used to indicate where each element starts in the buffer. See further notes about offset buffers in the description for `componentType`."
+      },
+      "componentType": {
+          "enum": [
+              "BOOLEAN",
```

Comment:
What if we have an `ARRAY[BOOLEAN]` -- is that an allowed combination, and if so, does that mean the offset buffers index _bits_ of the bit vector?

Reply:
Oh good point. I hadn't thought about that. I think each element would need to be padded to a byte, even if it's only using say 2 out of the 8 bits. Indexing by bits seems too extreme.

I wonder what Apache Arrow does here.

Reply:
> I think each element would need to be padded to a byte, even if it's only using say 2 out of the 8 bits.

Or maybe not, because then what's the point of using `BOOLEAN`...

I would just look at how Apache Arrow does this

Reply:
from Apache C++ API doc, it seems like they handle Bool array as a special case, similar to string array as well http://arrow.apache.org/docs/cpp/api/array.html#classarrow_1_1_boolean_array. 

This is also the case for std::vector<bool> in C++, which is handled differently compared to other type, where one element is only a bit.

Reply:
I guess we can specify that for the special case like ARRAY[BOOLEAN], offset array will index _bits_ of the bit vector. However, the first value in the offset array will index _bytes_ of the buffer. The begin and end of the boolean array should be a multiple of 8 bytes (for alignment).

Reply:
For ARRAY[BOOLEAN], I think `componentsPerElement` will be indexed by _bit_ instead of _byte_, but the offset of bufferview is still multiple of 8 bytes for alignment though.

![IMG-0341](https://user-images.githubusercontent.com/20376598/96193860-238c6500-0f17-11eb-901c-a45713d16a8c.jpg)

Reply:
Moving this discussion here: https://github.com/CesiumGS/3d-tiles-next/issues/41#issuecomment-710121233